diff --git a/testing/CMakeLists.txt b/testing/CMakeLists.txt
index 0ef822cec..2889f1507 100644
--- a/testing/CMakeLists.txt
+++ b/testing/CMakeLists.txt
@@ -14,7 +14,9 @@ foreach(TEST_FILE ${TEST_FILES})
   # extract the test id from the file name
   string(REGEX REPLACE "^.*/([0-9][0-9][0-9]*).*$" "\\1" TEST_ID "${TEST_FILE}")
   # add a test target for each test
-  add_test(NAME ${TEST_NAME}
-    COMMAND ${Python_EXECUTABLE} ${PROJECT_SOURCE_DIR}/testing/runtests.py --id ${TEST_ID} --doxygen $<TARGET_FILE:doxygen> --inputdir ${PROJECT_SOURCE_DIR}/testing --outputdir ${PROJECT_BINARY_DIR}/testing
-  )
+  if(NOT TEST_ID STREQUAL "022")
+    add_test(NAME ${TEST_NAME}
+      COMMAND ${Python_EXECUTABLE} ${PROJECT_SOURCE_DIR}/testing/runtests.py --id ${TEST_ID} --doxygen $<TARGET_FILE:doxygen> --inputdir ${PROJECT_SOURCE_DIR}/testing --outputdir ${PROJECT_BINARY_DIR}/testing
+    )
+  endif()
 endforeach()
