diff --git a/src/latexgen.cpp b/src/latexgen.cpp
index 739d29c7c..a2e7e5dc6 100644
--- a/src/latexgen.cpp
+++ b/src/latexgen.cpp
@@ -81,8 +81,13 @@ void LatexCodeGenerator::codify(const QCString &str)
     char c = 0;
     //char cs[5];
     int tabSize = Config_getInt(TAB_SIZE);
+#ifdef __MVS__
+    static char *result = NULL;
+    static int lresult = 0;
+#else
     static THREAD_LOCAL char *result = nullptr;
     static THREAD_LOCAL int lresult = 0;
+#endif
     if (m_hide) // only update column count
     {
       m_col = updateColumnCount(p,m_col);
