diff --git a/deps/spdlog/include/spdlog/details/os-inl.h b/deps/spdlog/include/spdlog/details/os-inl.h
index b9bab53ce..06989be5c 100644
--- a/deps/spdlog/include/spdlog/details/os-inl.h
+++ b/deps/spdlog/include/spdlog/details/os-inl.h
@@ -286,7 +286,7 @@ SPDLOG_INLINE int utc_minutes_offset(const std::tm &tm)
     return offset;
 #else
 
-#    if defined(sun) || defined(__sun) || defined(_AIX) || (!defined(_BSD_SOURCE) && !defined(_GNU_SOURCE))
+#    if defined(sun) || defined(__sun) || defined(_AIX) || (!defined(_BSD_SOURCE) && !defined(_GNU_SOURCE)) || defined(__MVS__)
     // 'tm_gmtoff' field is BSD extension and it's missing on SunOS/Solaris
     struct helper
     {
@@ -363,7 +363,7 @@ SPDLOG_INLINE size_t _thread_id() SPDLOG_NOEXCEPT
 // Return current thread id as size_t (from thread local storage)
 SPDLOG_INLINE size_t thread_id() SPDLOG_NOEXCEPT
 {
-#if defined(SPDLOG_NO_TLS)
+#if defined(SPDLOG_NO_TLS) || defined (__MVS__)
     return _thread_id();
 #else // cache thread id in tls
     static thread_local const size_t tid = _thread_id();
