diff --git a/testing/runtests.py b/testing/runtests.py
index b21762b91..167cfa456 100755
--- a/testing/runtests.py
+++ b/testing/runtests.py
@@ -4,6 +4,11 @@ from __future__ import print_function
 import argparse, glob, itertools, re, shutil, os, sys
 import subprocess
 import shlex
+import logging
+
+
+# Set logging level to DEBUG
+logging.basicConfig(level=logging.DEBUG)
 
 config_reg = re.compile(r'.*\/\/\s*(?P<name>\S+):\s*(?P<value>.*)$')
 bkmk_reg = re.compile(r'.*bkmkstart\s+([A-Z][A-Z][A-Z][A-Z][A-Z][A-Z][A-Z][A-Z][A-Z][A-Z]).*')
